vespa_analysis/vespa_eval.py

```diff
@@ -0,0 +1,159 @@
+    fasta_dict = dict()
+    for record in fasta_parsed:
+        fasta_dict[record.id] = str(record.seq)
+    return fasta_dict
+
+def run_vespa(fasta_file,cache_path):
+    cwd = os.getcwd()
+    os.system('vespa '+fasta_file+' --prott5_weights_cache '+cache_path+' --output_classes')
+    vespa_out = cwd+'/vespa_run_directory/output'
+    return vespa_out
+
+def run_mmseq(gen_fasta,ref_fasta,gen_name,threads):
+    cwd = os.getcwd()
+    os.system('mkdir -p '+gen_name)
+    os.chdir(gen_name)
+    os.system('mmseqs createdb '+gen_fasta+' '+gen_name+'_db')
+    os.system('mmseqs createdb '+ref_fasta+' '+ref_name+'_db')
+    os.system('mmseqs search '+gen_name+'_db '+ref_name+'_db res tmp --threads '+str(threads))
+    os.system('mmseqs align '+gen_name+'_db '+ref_name+'_db res '+gen_name+'_aln -a')
+    os.system('mmseqs convertalis '+gen_name+'_db '+ref_name+'_db '+gen_name+'_aln '+gen_name+'_aln.tab --format-output query,target,qaln,taln')
+    os.system('cp '+gen_name+'_aln.tab '+cwd)
+    os.chdir(cwd)
+    os.system('rm -r '+gen_name)
+    mmseq_out = cwd+'/'+gen_name+'_aln.tab'
+    return mmseq_out
+
+def generate_msa(ref_fasta,gen_fasta,threads):
+    os.system('mkdir -p vespa_msa')
+    os.system('cat '+ref_fasta+' '+gen_fasta+' > ref_gen.fasta')
+    os.system('clustalo -i ref_gen.fasta --threads='+str(threads)+' --iter=2 --full-iter -o vespa_msa/ref_gen_msa.fasta -v --force')
+    ref_gen_msa = 'vespa_msa/ref_gen_msa.fasta'
+    return ref_gen_msa
+
+## Function for Pairwise Mapping VESPA scores to Mutants ##
+def pair_map (gen_dict,ref_dict,mmseq_out,vespa_out,threads):
+    ref_id = ''
+    ref_seq = ''
+    gen_id = ''
+    gen_seq = ''
+
+    with open ('vespa_pair_cmd.list','w') as cmd:
+        for i in range(len(gen_dict)):
+            ref_id = list(ref_dict.keys())[i]
+            ref_seq = ref_dict[ref_id]
+            gen_id = list(gen_dict.keys())[i]
+            gen_seq = gen_dict[gen_id]
+            cmd.write('python vespa_pair.py --mmseq_out '+mmseq_out+' --vespa_out '+vespa_out+'/'+str(i)+'.csv --ref_id '+ref_id+' --ref_seq '+ref_seq+' --gen_id '+gen_id+' --gen_seq '+gen_seq+'\n')
+
+    os.system('parallel -j '+str(threads)+' < vespa_pair_cmd.list')
+
+## Function for MSA Mapping VESPA scores and Conservation Classes ##
+def msa_map (ref_fasta,gen_fasta,threads):
+    ref_gen_msa = generate_msa(ref_fasta,gen_fasta,threads)
+    msa_dict = fasta2dict(ref_gen_msa)
+    cons_fasta = cwd+'/vespa_run_directory/conspred_class.fasta'
+    cons_dict = fasta2dict(cons_fasta)
+    mapped_scores = cwd+'/vespa_scores'
+
+    with open ('vespa_msa_cmd.list','w') as cmd:
+        for i in range(len(gen_dict)):
+            ref_id = list(ref_dict.keys())[i]
+            gen_id = list(gen_dict.keys())[i]
+            ref_aln = msa_dict[ref_id]
+            gen_aln = msa_dict[gen_id]
+            cons_class = cons_dict[ref_id]
+            vespa_map_out = cwd+'/vespa_scores/'+gen_id+'_vespa_scores.csv'
```

Comment:
a better way to handle paths is to use `os.path.join` instead. example: `os.path.join(cwd, 'vespa_scores', gen_id+'_vespa_scores.csv').

because if anyone used this code they might pass an extra '/' that might crash the code.